src/types.ts

```diff
@@ -350,13 +351,22 @@ export type Subscribe = (...args: SubscribeArgs) => {
   unsubscribe: () => void
 };
 
+export type PrimitiveValue = string | number | boolean | null;
+export type ArrayValue = PrimitiveValue[] | JSONValue[];
+export type PrimitiveOrArrayValue = PrimitiveValue | ArrayValue;
+
+export type JSONValue = PrimitiveOrArrayValue | {
+  [key: string]: PrimitiveOrArrayValue | JSONValue;
+};
```

Comment:
Not sure if I've got this correct. `genomeBrowser.switch` is not happy in `send.ts` if the `JSONValue` type is similar to that in `ensembl-client`, so had to add in `PrimitiveOrArrayValue` as well.

Reply:
Peregrine is also capable of handling json objects... Don't know if Dan intends to; but his function parses objects as well.